lmdeploy/pytorch_poc/engine/engine.py

```diff
@@ -152,9 +153,12 @@ def __init__(self,
                  model_path: str,
                  scheduler_config: SchedulerConfig = None,
                  cache_config: CacheConfig = None) -> None:
-        hf_model = AutoModelForCausalLM.from_pretrained(model_path)
-        self.patched_model = patch(hf_model, ['context']).cuda()
+        with LoadNoInit():
+            hf_model = AutoModelForCausalLM.from_pretrained(
+                model_path, torch_dtype=torch.float16)
```

Comment:
torch_dtype can be read from `AutoConfig.from_pretrained`

Reply:
We can set it to 'auto' to make it read from the config https://github.com/huggingface/transformers/blob/26a38a9033d3fc247aa0c3d9e7afb665f73ec5d5/src/transformers/modeling_utils.py#L2821